### PR TITLE
[IMP] barcode: update default nomenclature doc

### PR DESCRIPTION
**Pull request by brse-odoo**
This PR is an [IMP] to improve/update the `barcode_nomenclature` doc in Odoo's Barcode documentation, specific to 17.0.

Here's what I did:
- Changed title from `Overview` to `Default barcode nomenclature` to accurately represent body of doc
- Overhauled information, structure, content, and added table at bottom
- Replaced screenshots with 17.0-specific images

### CODE REVIEW

**Comment by brse-odoo** [comment]
Hi @odoo/barcode-doc-review, this PR is ready for content peer review when one of you has a chance. Thank you!

**Comment by brse-odoo** [comment]
> Hi @odoo/barcode-doc-review, this PR is ready for content peer review when one of you has a chance. Thank you!

Hi @odoo/barcode-doc-review this PR is ready for review, thanks!

**Comment by Felicious** [comment]
@brse-odoo Thanks for bumping this! I'll review this after the standup!

**Review by Felicious — APPROVED** [review]
Love this update of the barcode nomenclature doc, @brse-odoo ! It's really clear and the time you took to input the nomenclature chart, line by line, is truly appreciated.

I barely had anything to point out, so after you address the few things I mentioned in my review, this doc is ready for the next round of review (:

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:39]: Thank you!

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:43]: I'm seeing the same thing now. I've updated the two paragraphs including this instruction and the one from your comment below to take out this part and instead instruct them to access it the other way.

Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:47]: This, along with the above comment, has been updated to have the reader access the page a different way, since this one seems to not work in 17.0.

Suggestion implemented ✅

**Comment by brse-odoo** [comment]
Thank you @Felicious for the review! Alright @odoo/us-doc-review, this PR is ready for review when you have a chance. Thank you!

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:96]: Do you want to add a quick mention/explanation of the 'Encoding' field, as well? Just to complete the pop-up window field explanations?

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:100]:
```suggestion
The :guilabel:`Barcode Pattern` represents how the sequence of letters or numbers is recognized by
the system to contain information about the product. Odoo follows the sequential order of this
table, and uses the first rule it matches, based on the sequence. 
```

Comment (ksc-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:100]: just to follow the order of things as they appear on the pop-up

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:107]:
```suggestion
   They are indicated by **{NNN}**, where 'N' defines where the number's digits are encoded. *Floats*
   are also supported with the decimals, indicated by 'D', such as **{NNNDD}**.
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:109]:
```suggestion
   In this case, the barcode field on the associated records **must** show these digits as zeros.
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:112]:
```suggestion
After filling in the information, click the :guilabel:`Save & New` button to save the rule, and
instantly start creating another rule. Or, click :guilabel:`Save & Close` to save the rule, and
return to the table of rules.
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:116]:
```suggestion
   When the :guilabel:`Barcode Pattern` field contains `.*`, it means it can contain **any**
   number of characters - those characters being **any** number.
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:123]:
```suggestion
The table below contains Odoo's list of Default Nomenclature rules. Barcode patterns are written in
```

**Review by ksc-odoo — CHANGES_REQUESTED** [review]
hey @brse-odoo -- just finished my Final Review of this PR. Great work. Once you address my feedback, and make the necessary adjustments, feel free to tag me again for another quick look. Thanks! 👍

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:23]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:35]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:57]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:77]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:100]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:107]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:109]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:112]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:48]: Added in an explanation for each of those fields. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:44]: I don't want to add a screenshot of the whole page, because that's what the table at the bottom of the doc is for. So, as a compromise, I added in a ref link to the table at the bottom in the paragraph that closes this section. Then, I added in a screenshot of the fields at the top of the page that are explained in this section.

Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:44]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:23]:
```suggestion
   official provider of |UPC|/|EAN| and GS1 |GTINs| in the world.
```
just sounds a bit cleaner with this phrasing

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:54]:
```suggestion
another when trying to match a rule with the other encoding.

The other available options for this field are: :guilabel:`Never`, :guilabel:`EAN-13 to UPC-A`, and
:guilabel:`UPC-A to EAN-13`.
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:62]:
```suggestion
The last field at the top of the page is the :guilabel:`Is GS1 Nomenclature` field. For the
:guilabel:`Default Nomenclature`, this field should remain unchecked. This is because the
:guilabel:`Default Nomenclature` uses |UPC| and |EAN| encoding, *not* GS1 encoding.
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:120]:
```suggestion
The :guilabel:`Sequence` field represents the priority of the rule; meaning the smaller the value, the
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:123]:
```suggestion
The barcode :guilabel:`Type` field represents different classifications of information that can be
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:127]:
```suggestion
The :guilabel:`Encoding` field specifies which encoding the barcode uses; this rule **only** applies if the
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:128]:
```suggestion
barcode uses this specific encoding. The available :guilabel:`Encoding` options are: **EAN-13**,
```
if these options appear on the UI (if chosen), wouldn't a GUI label be used in this instance? What are your thoughts?

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:131]:
```suggestion
The :guilabel:`Barcode Pattern` field represents how the sequence of letters or numbers is recognized by
```

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:157]:
```suggestion
The table below contains Odoo's list of :guilabel:`Default Nomenclature` rules. Barcode patterns are written in
```

**Review by ksc-odoo — APPROVED** [review]
hey @brse-odoo -- thanks for making all those suggested changes so quickly. really appreciate that. since I only have a handful of other suggestions that require your attention, I'll go ahead and approve this for now. So, once you implement all the necessary changes from this review, you can feel free to tag this for Tech Review. Thanks! 👍

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:23]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:54]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:62]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:120]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:123]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:127]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:128]: I went back and forth between the two, but yeah I agree with you - so I've made the change. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:131]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:157]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
> hey @brse-odoo -- thanks for making all those suggested changes so quickly. really appreciate that. since I only have a handful of other suggestions that require your attention, I'll go ahead and approve this for now. So, once you implement all the necessary changes from this review, you can feel free to tag this for Tech Review. Thanks! 👍

Thank you @ksc-odoo for the review. Alright @samueljlieber, this PR is ready for tech review when you get the chance. Thank you!

**Review by samueljlieber — APPROVED** [review]
Hi @brse-odoo! Awesome job improving this doc, everything looks really great. I left a few suggestions for improvement below, but overall not many changes needed. Please see below and feel free to tag ZST once you are ready, approving now. Thanks!

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:75]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:213]: Good idea! Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:152]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature/barcode-nomenclature-new-rule-popup.png
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature/barcode-nomenclature-new-rule-popup.png:1]: I changed the screenshot because it does look a lot better the way you suggested! Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:152]: Ah, that's interesting thank you for pointing that out! 🙏

**Comment by brse-odoo** [comment]
> Hi @brse-odoo! Awesome job improving this doc, everything looks really great. I left a few suggestions for improvement below, but overall not many changes needed. Please see below and feel free to tag ZST once you are ready, approving now. Thanks!

Hi @samueljlieber, thank you for the quick review! @StraubCreative This PR is ready for your final lookover before merge, thank you!

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:51]: The formatting broke here. I suggest to remove the vertical bar symbols `|` from the gui

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst:60]: Same here, the rendered HTML is not picking up the shortcode markup due to the back ticks in the gui.

**Review by StraubCreative — APPROVED** [review]
Hi @brse-odoo 

I think this is a great rewrite that explains UPC/EAN very well and in a clear, instructional way for use in Odoo. Nice job!

I have two necessary revisions for you below where the RST shortcodes didn't render correctly in the HTML. It's a super quick fix where we should just remove the shortcode markup from the guis (specifically, the vertical pipe symbols: `|`).

After those two changes you can merge, thank you.
...
@robodoo delegate=brse-odoo

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst]: Removed vertical bar symbols - Suggestion implemented ✅

File: content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst]: Removed vertical bar symbols - Suggestion implemented ✅

**Comment by brse-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
